dm/worker/server.go

```diff
@@ -491,6 +489,8 @@ func (s *Server) doClose() {
 
 // Close closes the RPC server, this function can be called multiple times.
 func (s *Server) Close() {
+	s.closeMu.Lock()
+	defer s.closeMu.Unlock()
```

Comment:
this may not work for this:
- `Start()` try `closeMu.Lock()`, but hasn't success
- `dm-worker` received signal to close
- `Close()` in signal routine try lock and success, and finish close
- `Start()` continue, and start success
- `dm-worker` keeps running

Reply:
seems this case doesn't exist, since `inited` is set to true on `close`

Reply:
> seems this case doesn't exist, since inited is set to true on close

it do exist, since `s.closed = true`, we didn't set `inited` to true in this case, so worker keeps running